### PR TITLE
Add regression test for all-NA BADM carbon pool filtering

### PR DESCRIPTION
**Pull request by shreyannandanwar**
### Summary

This PR adds a regression test covering the case where all BADM carbon pool values are `NA`.

### Changes

- Add a test to `test-IC_BADM_Utilities.R` verifying that all-NA carbon pool
  entries are handled cleanly
  
### Issue

#3799

### Tests

- Added test to `test-IC_BADM_Utilities.R`
- All existing tests pass

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I agree that PEcAn Project may distribute my contribution under any or all of
	- the same license as the existing code,
	- and/or the BSD 3-clause license.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: modules/data.land/tests/testthat/test-IC_BADM_Utilities.R [hunk 1]
```diff
@@ -19,6 +19,30 @@ test_that("Read.IC.info.BADM falls back to L1 and ALL if no L2 data", {
...
+  entries <- data.frame(
+    GROUP_ID = c("A", "B", "C"),
+    LAT = c(42, 43, 44),
+    LON = c(-72, -73, -74),
+    UNIT = c("kg/m2", "kg/m2", "kg/m2"),
+    leafC = c(NA, NA, NA),
+    stemC = c(NA, NA, NA),
+    rootC = c(NA, NA, NA),
+    soilC = c(NA, NA, NA),
+    stringsAsFactors = FALSE
```
Comment (S1DDHEY) [inline modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: For these test entries, it would be better to use the documented `BADM` long-format (`VARIABLE` / `DATAVALUE`) structure rather than introducing ad-hoc columns.

Those can be found here https://pecanproject.github.io/package-documentation/develop/PEcAn.data.land/reference/BADM.html#format

This will be helpful while testing using the function itself.

File: modules/data.land/tests/testthat/test-IC_BADM_Utilities.R [hunk 2]
```diff
@@ -19,6 +19,30 @@ test_that("Read.IC.info.BADM falls back to L1 and ALL if no L2 data", {
...
+  )
+  
+  # Run the same filtering logic used internally
+  ind <- apply(entries[, 5:8], 1, function(x) all(is.na(x)))
+  filtered <- entries[-which(ind), ]
```
Comment (S1DDHEY) [inline modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: Using the function directly is recommended instead of re-implementing ad-hoc logic, so the test validates the real behavior.

**Review by S1DDHEY — CHANGES_REQUESTED** [review]

**Comment by shreyannandanwar** [comment]
Thanks for the feedback!
You're right - I was testing the internal logic rather than the actual function behavior. I will be updating the test

**Review by S1DDHEY — APPROVED** [review]

File: modules/data.land/tests/testthat/test-IC_BADM_Utilities.R [hunk 3]
```diff
@@ -19,6 +19,23 @@ test_that("Read.IC.info.BADM falls back to L1 and ALL if no L2 data", {
...
+  result <- Read.IC.info.BADM(lat, lon)
+  
+  expect_s3_class(result, "data.frame")
+  
+  if (nrow(result) > 0) {
+    carbon_cols <- c("AGB", "soil_organic_carbon_content", "litter_carbon_content", "root_carbon_content")
+    
+    has_data <- apply(result[, carbon_cols], 1, function(x) any(!is.na(x)))
+    expect_true(all(has_data), info = "All rows should have at least one non-NA carbon value")
+  }
```
Comment (infotroph) [inline modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: `if` is undesirable here:
* The result is deterministic, but this makes it look like there could be variation
* If a future regression did make this query return 0 rows, that should be an error not a a silent pass.

```suggestion
  carbon_cols <- c("AGB", "soil_organic_carbon_content", "litter_carbon_content", "root_carbon_content")  
  has_data <- apply(result[, carbon_cols], 1, function(x) any(!is.na(x)))
  expect_true(all(has_data), info = "All rows should have at least one non-NA carbon value")
```

Comment (infotroph) [reply modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: Note also that technically this isn't testing that NAs are _removed_, just that this specific query doesn't _return_ any. Knowing that 3 all-NA record did exist to be removed by the filter requires digging around in the source data.

**Review by infotroph — CHANGES_REQUESTED** [review]

File: modules/data.land/tests/testthat/test-IC_BADM_Utilities.R (hunk 3 above)
Comment (shreyannandanwar) [reply modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: Thanks @infotroph, 
my previous approach didn't actually prove all-NA rows were being filtered out. So, I made a new one with controlled data: 
I created a test dataset with a known all-NA row, applied the filtering logic. I also tested edge cases - like all rows NA and no NA at all - to ensure it works reliably. This way, I’m testing the fix directly.

Comment (infotroph) [reply modules/data.land/tests/testthat/test-IC_BADM_Utilities.R]: I appreciate you trying here, but this update is a big step backwards -- it goes back to not actually calling `IC_BADM_Utilities`.  

Remember that when you're testing the task is always to prove correct behavior of _the code that's actually implemented_. And since this is a regression test for an issue fixed in a specific known patch, the test-driven development mantra applies as well: If you haven't seen the test _fail_ against the buggy code (i.e. the version of `IC_BADM_Utilities` from before PR #3801) then seeing it _pass_ against the patched code doesn't prove the bug is fixed.

**Comment by infotroph** [comment]
@shreyannandanwar Are you interested in working more on this? My comments in https://github.com/PecanProject/pecan/pull/3804#discussion_r2855021682 still apply: The output you test on needs to be generated by the function you're trying to test.

**Comment by shreyannandanwar** [comment]
@infotroph 
Yes, I will work on it. And provide the solution for review.

**Comment by shreyannandanwar** [comment]
I've updated the regression test so that it exercises `Read.IC.info.BADM()` directly rather than reimplementing the filtering logic inside the test. The assertions now validate the function output and verify that all-NA carbon pool rows are not returned.
